### PR TITLE
Extract setting up ec2 linux and checkout to be composite actions

### PR DESCRIPTION
**Pull request by ZainRizvi**
Step 1 to gradually change our generated workflow scripts to be more compatible with reusable workflows.

This PR extracts three pieces of functionality into composite actions:
- Checking out the branch
- Setting up EC2 linux 
- Building the linux binary

It also regenerates the workflows to use these new composite actions

This is not a complete list of things we'd like to extract. Keeping this PR small to get early feedback

### CODE REVIEW

File: .github/actions/common-setup-ec2-linux/action.yml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: Setup EC2 Linux
```
Comment (seemethere) [inline .github/actions/common-setup-ec2-linux/action.yml:1]: Is this not already covered by `.github/actions/setup-linux`?

**Review by seemethere — APPROVED** [review]
Mostly LGTM, just had a question about redundancy for the setup step

**Review by malfet — CHANGES_REQUESTED** [review]
Hmm, can we unify `common-checkout` and `checkout-pytorch` actions?
And `setup-linux` and `common-setup-ec2-linux`?

File: .github/actions/common-setup-ec2-linux/action.yml (hunk 1 above)
Comment (ZainRizvi) [reply .github/actions/common-setup-ec2-linux/action.yml:1]: This composite action calls the `.github/actions/setup-linux` action and does some extra work before & after it

**Comment by ZainRizvi** [comment]
> Hmm, can we unify `common-checkout` and `checkout-pytorch` actions? And `setup-linux` and `common-setup-ec2-linux`?

I do agree that it would be good to unify the logic for binary build actions and regular actions. (note that `common-setup-ec2-linux` invokes `setup-linux`).  

However, at this point I’m hesitant to change the workflow logic to unify more actions since that’s going to make it much harder to verify that the workflow functionality isn’t modified.  For example, the `common-checkout` and `checkout-pytorch` functionality look very different and depend on different actions in turn.

There are a ton more actions that are going to be extracted out from the jinja template, so I’d rather split this up into two projects:
1. Extract out the action files from the common codgen
2. Refactor the action files to reduce duplication

**Comment by malfet** [comment]
Sorry, categorisation should have been: missed signal. @ZainRizvi please add `ciflow/trunk` to PRs that modify binary build rules (and/or, as next step, `ciflow/binaries`)

**Pull request closed without merging**